argus/backend/service/planner_service.py

```diff
@@ -438,3 +448,88 @@ def resolve_plan(self, plan_id: str | UUID) -> list[dict[str, Any]]:
                 ent["group"] = group.pretty_name or group.name
 
         return mapped
+    
+    def trigger_jobs(self, payload: PlanTriggerPayload) -> bool:
+
+        release_name = payload.get("release")
+        plan_id = payload.get("plan_id")
+        version = payload.get("version")
+
+        condition_set = (bool(release_name), bool(plan_id), bool(version))
+
+        match condition_set:
+            case (True, False, False):
+                release = ArgusRelease.get(name=release_name)
+                filter_expr = { "release_id__eq": release.id }
+            case (False, True, False):
+                filter_expr = { "id__eq": plan_id }
+            case (False, False, True):
+                filter_expr = { "target_version__eq": version }
+            case _:
+                raise PlannerServiceException("No version, release name or plan id specified.", payload)
```

Comment:
this will be raised also when multiple are selected (like version and release). Is it intended? maybe it should be validated within `PlanTriggerPayload`?

Reply:
You are correct, after discussion we decided to also support (version, release) pair, but yes, previously that would be an exception and it was intended.

Reply:
Done